m3u8/model.py

```diff
@@ -137,35 +147,33 @@ def __init__(self, content=None, base_path=None, base_uri=None, strict=False):
         self._initialize_attributes()
         self.base_path = base_path
 
+
     def _initialize_attributes(self):
-        self.key = Key(base_uri=self.base_uri, **self.data['key']) if 'key' in self.data else None
         self.segments = SegmentList([ Segment(base_uri=self.base_uri, **params)
                                       for params in self.data.get('segments', []) ])
-
+        self.keys = [ key for key in set(segment.key for segment in self.segments) ]
         for attr, param in self.simple_attributes:
             setattr(self, attr, self.data.get(param))
 
         self.files = []
-        if self.key:
-            self.files.append(self.key.uri)
+        for key in self.keys:
```

Comment:
We could try to do list comprehension here and then a filter `L = filter(None, originalList)` and apply unique.